src/librustc_typeck/diagnostics.rs

```diff
@@ -851,7 +851,7 @@ optional namespacing), a dereference, an indexing expression or a field
 reference.
 
 More details can be found here:
-https://doc.rust-lang.org/reference.html#lvalues-rvalues-and-temporaries
+https://doc.rust-lang.org/reference/expressions.html#lvalues-rvalues-and-temporaries
```

Comment:
I believe the correct fix here is to change https://github.com/rust-lang/rust/blob/master/src/tools/tidy/src/style.rs#L55 to properly handle URLs in non-comment lines. I haven't looked into the exact fix, though. It might be non-trivial.